x86/src/Data/Macaw/X86/Semantics.hs

```diff
@@ -2450,23 +2450,51 @@ def_pselect mnem op sz = defBinaryLV mnem $ \l v -> do
 -- PEXTRW Extract word
 
 -- | PINSRW Insert word
-exec_pinsrw :: Location (Addr ids) XMMType -> BVExpr ids 16 -> Word8 -> X86Generator st ids ()
-exec_pinsrw l v off = do
+exec_pinsrx :: (1 <= w, 1 <= c)
+            => NatRepr w
+            -> NatRepr c
```

Comment:
Could we get a comment explaining what `w` and `c` are here?

Are the mask values just taken from the ISA manual?

Reply:
I've added some Haddock comments to those parameters.

The ISA manual specifies to use the 1, 2, 3, or 4 least significant bits of the immediate (depending on the instruction / whether the destination is MMX or XMM for `pinsrw`). I can add comments explaining the masks in this light if you'd like. I could also a binary literal which makes the masks more apparent, but this would require enabling the `BinaryLiterals` extension.